src/commands/cmd_zset.cc

```diff
@@ -966,24 +966,55 @@ class CommandZRevRangeByScore : public CommandZRangeGeneric {
 class CommandZRank : public Commander {
  public:
   explicit CommandZRank(bool reversed = false) : reversed_(reversed) {}
+
+  Status Parse(const std::vector<std::string> &args) override {
+    if (args.size() > 4) {
+      return {Status::RedisParseErr, errWrongNumOfArguments};
+    }
+
+    // skip the <CMD> <key> <member> and parse remaining optional arguments
+    CommandParser parser(args, 3);
+    while (parser.Good()) {
+      if (parser.EatEqICase("withscore") && !with_score_) {
+        with_score_ = true;
+      } else {
+        return parser.InvalidSyntax();
+      }
+    }
+
+    return Commander::Parse(args);
+  }
+
   Status Execute(Server *svr, Connection *conn, std::string *output) override {
     int rank = 0;
+    double score = 0.0;
     redis::ZSet zset_db(svr->storage, conn->GetNamespace());
-    auto s = zset_db.Rank(args_[1], args_[2], reversed_, &rank);
+    auto s = zset_db.Rank(args_[1], args_[2], reversed_, &rank, &score);
     if (!s.ok()) {
       return {Status::RedisExecErr, s.ToString()};
     }
 
```

Comment:
seems Withscore will change the return value in `NotFound` case:

```
127.0.0.1:6379>  ZADD myzset 1 "one"
(integer) 1
127.0.0.1:6379>  ZADD myzset 2 "two"
(integer) 1
127.0.0.1:6379> ZADD myzset 3 "three"
(integer) 1
127.0.0.1:6379> ZRANK myzset "four"
(nil)
127.0.0.1:6379> ZRANK myzset "four" WITHSCORE
(error) ERR wrong number of arguments for 'zrank' command
127.0.0.1:6379> 
```

Reply:
which redis version are you using? the option was introduced in Redis 7.2 RC

Reply:
Oh I see, I use 7.0 here, after verifying using unstable code, this is ok for me